code/game/atoms_movable.dm

```diff
@@ -1291,6 +1291,11 @@
 /atom/movable/hitby(atom/movable/hitting_atom, skipcatch, hitpush = TRUE, blocked, datum/thrownthing/throwingdatum)
 	if(HAS_TRAIT(src, TRAIT_NO_THROW_HITPUSH))
 		hitpush = FALSE
+	var/mob/living/carbon/thrower = throwingdatum?.get_thrower()
+	if(HAS_TRAIT(thrower, TRAIT_NOGUNS) && isgun(hitting_atom)) //throw a gun at them when they don't expect it
```

Comment:
Is there no proc called by a thrown atom when it hits something? Ideally this should be there under the gun subtype, not performing a typecheck on everything that hits anything else.

Reply:
I've done something. Is it fine?